Tests/AnyCodableTests/AnyCodableTests.swift

```diff
@@ -171,14 +219,149 @@ class AnyCodableTests: XCTestCase {
+        let string3 = String(data: data3, encoding: .utf8)
+        XCTAssertEqual(string3, #"["1970-01-01T00:00:00.000Z"]"#)
+
+        let anyCodable4 = try AnyCodable.encodable(date, dateFormat: .date)
+        let expectedAnyCodable4: AnyCodable = "1970-01-01"
+        XCTAssertEqual(anyCodable4, expectedAnyCodable4)
+
+        let data4 = try JSONEncoder().encode([anyCodable4]) // Use array for swift 5.1 support
+        let string4 = String(data: data4, encoding: .utf8)
+        XCTAssertEqual(string4, #"["1970-01-01"]"#)
+    }
+    
+    func test_RefInit() {
+        var value = 0
+        let ref = Ref(get: {
+            value
+        }, set: {
+            value = $0
+        })
+        XCTAssertEqual(ref.wrappedValue, 0)
+        ref.wrappedValue = 1
+        XCTAssertEqual(ref.wrappedValue, 1)
+    }
+    
+    func test_RefConstant() {
+        let ref = Ref.constant(1)
+        XCTAssertEqual(ref.wrappedValue, 1)
+        ref.wrappedValue = 2
+        XCTAssertEqual(ref.wrappedValue, 1)
+    }
+    
+    func test_RefByKeyPath() {
+        let ref = Ref(DateFormatter(), \.dateFormat)
+        ref.wrappedValue = "yyyy-MM-dd"
+        XCTAssertEqual(ref.wrappedValue, "yyyy-MM-dd")
+        ref.wrappedValue = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        XCTAssertEqual(ref.wrappedValue, "yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+    }
+    
+    func test_description() {
+        XCTAssertEqual(AnyCodable(0).description, "0")
+        XCTAssertEqual(AnyCodable(0.0).description, "0.0")
+        XCTAssertEqual(AnyCodable(()).description, "nil")
+        XCTAssertEqual(AnyCodable("hello").description, "\"hello\"")
+        XCTAssertEqual(AnyCodable(true).description, "true")
+        XCTAssertEqual(AnyCodable(false).description, "false")
+        XCTAssertEqual(AnyCodable([1, 2, 3]).description, "[1, 2, 3]")
+        XCTAssertEqual(AnyCodable(["a": 1, "b": 2]).description, "[\"a\": 1, \"b\": 2]")
+    }
+    
+    func test_debugDescription() {
+        XCTAssertEqual(AnyCodable(0).debugDescription, "AnyCodable(0)")
+        XCTAssertEqual(AnyCodable(0.0).debugDescription, "AnyCodable(0.0)")
+        XCTAssertEqual(AnyCodable(()).debugDescription, "AnyCodable(nil)")
+        XCTAssertEqual(AnyCodable("hello").debugDescription, "AnyCodable(\"hello\")")
+        XCTAssertEqual(AnyCodable(true).debugDescription, "AnyCodable(true)")
+        XCTAssertEqual(AnyCodable(false).debugDescription, "AnyCodable(false)")
+        XCTAssertEqual(AnyCodable([1, 2, 3]).debugDescription, "AnyCodable([1, 2, 3])")
+        XCTAssertEqual(AnyCodable(["a": 1, "b": 2]).debugDescription, "AnyCodable([\"a\": 1, \"b\": 2])")
+    }
 }
 
-fileprivate struct Wrapper: Codable {
+private struct Wrapper: Codable {
     let value: AnyCodable
 }
+
+private struct EncodableStruct: Codable, Equatable {
+    var int = 1
+    var string = "hello"
+    var bool = true
+    var array = [1, 2, 3]
+    var dictionary = ["a": 1, "b": 2, "c": 3]
```

Comment:
```suggestion
    var dictionary = ["a": 1, "b": 2, "c": 3]
    var data = "hello".data(using: .utf8)
    var decimal = Decimal(10)
```
This should be the way to test the other two special cases in your `func encode(_ value: Encodable) throws -> AnyCodable` function. The first property works once the additional field is added to the relevant test cases above. The second property appears to expose a bug in the `encode` function because `Decimal` gets encoded as a `String` value which then will fail to decode as a `Decimal`.

Reply:
@mattpolzin Thank you very much, I'll check it all soon

Reply:
I've observed that certain Foundation types (URL, Decimal, Data) don't have optimal native encoding/decoding methods. While JSONEncoder/JSONDecoder offer custom encoding strategies for these types, I believe the AnyCodable encoder should also provide a flexible means to customize encoding strategies for any type. Hence, I've introduced the ValueEncodingStrategy.

Reply:
I checked your OpenAPIReflection cases, now there is .init(Encodable), but it throws, because it uses encoder under hood. I can replace `try encode()` with `try? encode() ?? .null`.

Reply:
Will `.init(Encodable)` suffice for https://github.com/mattpolzin/OpenAPIReflection/blob/main//Sources/OpenAPIReflection/AnyJSONCaseIterable.swift#L74..L74 where a value happens to be `Encodable` but the compiler may only know it as `Any`?

Unless I am missing something, I don't see your new `.init(Encodable)` in this PR; have you pushed it up yet? I'd be curious to see why it is necessary to encode something on initialization instead of just when actually encoding it. Perhaps because part of the point is to make sure we have something equatable... Is this an indication that backwards compatibility is not worth it?

Maybe your new type is functionally just as good, if only you didn't need to support initialization from `Any`. Though I would need to think more about whether that really does serve enough use cases (without supporting `Any`) because I use `AnyCodable` pretty extensively in downstream projects (for the reflection library, yes, but also in libraries of mine that support generating Swift types from OpenAPI documents and even testing OpenAPI documents for consistency in their examples using those generated Swift types).

Reply:
Hi, I apologize for the delayed response. I will return .init(Any) with your switch by type. In this case, encoding should occur rarely.